java/core/src/main/java/com/google/protobuf/CodedInputStream.java

```diff
@@ -2864,8 +2924,42 @@ public byte readRawByte() throws IOException {
         pos = tempPos + size;
         return Arrays.copyOfRange(buffer, tempPos, tempPos + size);
```

Comment:
Should this be using System.arraycopy as well? Also concerning that we're not catching/rethrowing exceptions here.